common/endian.h

```diff
@@ -623,6 +623,125 @@ inline void WRITE_BE_UINT24(void *ptr, uint32 value) {
 #define WRITE_UINT24(a,b) WRITE_BE_UINT24(a,b)
 #endif
 
+union SwapFloat {
+	float f;
+	uint32 u32;
+};
+
+STATIC_ASSERT(sizeof(float) == sizeof(uint32), Unexpected_size_of_float);
+
+inline float READ_LE_FLOAT32(const void *ptr) {
+	SwapFloat swap;
+	swap.u32 = READ_LE_UINT32(ptr);
+	return swap.f;
+}
+
+inline void WRITE_LE_FLOAT32(void *ptr, float value) {
+	SwapFloat swap;
+	swap.f = value;
+	WRITE_LE_FLOAT32(ptr, swap.u32);
+}
+
+inline float READ_BE_FLOAT32(const void *ptr) {
+	SwapFloat swap;
+	swap.u32 = READ_BE_UINT32(ptr);
+	return swap.f;
+}
+
+inline void WRITE_BE_FLOAT32(void *ptr, float value) {
+	SwapFloat swap;
+	swap.f = value;
+	WRITE_BE_FLOAT32(ptr, swap.u32);
+}
+
+#ifdef SCUMM_LITTLE_ENDIAN
+#define READ_FLOAT32(a) READ_LE_FLOAT32(a)
+#define WRITE_FLOAT32(a,b) WRITE_LE_FLOAT32(a,b)
+#else
+#define READ_FLOAT32(a) READ_BE_FLOAT32(a)
+#define WRITE_FLOAT32(a,b) WRITE_BE_FLOAT32(a,b)
+#endif
+
+#ifdef SCUMM_FLOAT_WORD_LITTLE_ENDIAN
+union SwapDouble {
+	double d;
+	uint64 u64;
+	struct {
+		uint32 low, high;
+	} u32;
+};
+#else
+union SwapDouble {
+	double d;
+	uint64 u64;
+	struct {
+		uint32 high, low;
+	} u32;
+};
+#endif
+
+STATIC_ASSERT(sizeof(double) == sizeof(uint64), Unexpected_size_of_double);
+
+inline double READ_LE_FLOAT64(const void *ptr) {
+	SwapDouble swap;
+	const uint8 *b = (const uint8 *)ptr;
+	swap.u32.low  = READ_LE_UINT32(b);
+	swap.u32.high = READ_LE_UINT32(b + 4);
+	return swap.d;
+}
+
+inline void WRITE_LE_FLOAT64(void *ptr, double value) {
+	SwapDouble swap;
+	swap.d = value;
+	uint8 *b = (uint8 *)ptr;
+	WRITE_LE_FLOAT32(b,     swap.u32.low);
+	WRITE_LE_FLOAT32(b + 4, swap.u32.high);
+}
+
+inline double READ_BE_FLOAT64(const void *ptr) {
+	SwapDouble swap;
+	const uint8 *b = (const uint8 *)ptr;
+	swap.u32.high = READ_BE_UINT32(b);
+	swap.u32.low  = READ_BE_UINT32(b + 4);
+	return swap.d;
+}
+
+inline void WRITE_BE_FLOAT64(void *ptr, double value) {
+	SwapDouble swap;
+	swap.d = value;
+	uint8 *b = (uint8 *)ptr;
+	WRITE_BE_FLOAT32(b,     swap.u32.high);
+	WRITE_BE_FLOAT32(b + 4, swap.u32.low);
+}
+
+inline double READ_FPA_FLOAT64(const void *ptr) {
```

Comment:
These two just seem to be duplicates of `READ_LE_FLOAT64` and `WRITE_LE_FLOAT64`.  If having the same functions available under two different names is meaningful, I'd suggest having one of the names simply calling the other to reduce code duplication.  After all, the functions are inline...

Reply:
This function uses big endian word order and little endian byte order, so isn't identical to `READ_LE_FLOAT64` or `WRITE_LE_FLOAT64`. Some background can be found here: https://github.com/libsdl-org/SDL/pull/5607#issuecomment-1117546113

Reply:
Ah, yes, I see now it's "PDP endian".  Is it ever used?

Reply:
Not at the moment - I originally added it in case I needed to maintain save file compatibility, but it might still be useful in the future when reading from game data files.